docs/src/design/chiplets/memory.md

```diff
@@ -170,111 +170,171 @@ The layout of Miden VM memory table is shown below:
+$$
+
 To simplify description of constraints, we'll define two variables $n_0$ and $n_1$ as follows:
 
 $$
-n_0 = \Delta c \cdot t' \\
+n_0 = \Delta ctx \cdot t' \\
 n_1 = \Delta a \cdot t'
 $$
 
-Where $\Delta c = c' - c$ and $\Delta a = a' - a$.
+Where $\Delta ctx = ctx' - ctx$ and $\Delta a = a' - a$.
 
 To make sure the prover sets the value of column `t` correctly, we'll need to impose the following constraints:
 
 >$$
-n_0^2 - n_0 = 0 \text{ | degree} = 4
+f_{mem\_nl} \cdot (n_0^2 - n_0) = 0 \text{ | degree} = 7
 $$
 
 >$$
-(1 - n_0) \cdot  \Delta c = 0 \text{ | degree} = 3
+f_{mem\_nl} \cdot (1 - n_0) \cdot  \Delta ctx = 0 \text{ | degree} = 7
 $$
 
 >$$
-(1 - n_0) \cdot (n_1^2 - n_1) = 0 \text{ | degree} = 6
+f_{mem\_nl} \cdot (1 - n_0) \cdot (n_1^2 - n_1) = 0 \text{ | degree} = 9
 $$
 
 >$$
-(1 - n_0) \cdot (1 - n_1) \cdot \Delta a = 0 \text{ | degree} = 5
+f_{mem\_nl} \cdot (1 - n_0) \cdot (1 - n_1) \cdot \Delta a = 0 \text{ | degree} = 8
 $$
 
 The above constraints guarantee that when context changes, $n_0 = 1$. When context remains the same but address changes, $(1 - n_0) \cdot n_1 = 1$. And when neither the context nor the address change, $(1 - n_0) \cdot (1 - n_1) = 1$.
 
-To enforce the values of the selector columns, we first require that they both contain only binary values.
+We enforce that the `rw`, `ew`, `idx0` and `idx1` contain binary values.
 
 >$$
-s_0^2 - s_0 = 0 \text{ | degree} = 2
+f_{mem} \cdot (rw^2 - rw) = 0 \text{ | degree} = 5
 $$
 
 >$$
-s_1^2 - s_1 = 0 \text{ | degree} = 2
+f_{mem} \cdot (ew^2 - ew) = 0 \text{ | degree} = 5
 $$
 
-Then we require that $s_1$ is always set to $1$ during read operations when the context and address did not change and to $0$ in all other cases.
-
 >$$
-(1 - n_0) \cdot (1 - n_1) \cdot s'_0 \cdot (1 - s'_1) = 0 \text{ | degree} = 6
+f_{mem} \cdot (idx0^2 - idx0) = 0 \text{ | degree} = 5
 $$
 
 >$$
-(n_0 + (1 - n_0) \cdot n_1  + (1 - s'_0)) \cdot s'_1 = 0 \text{ | degree} = 5
+f_{mem} \cdot (idx1^2 - idx1) = 0 \text{ | degree} = 5
 $$
 
-The first constraint enforces that `s_1` is $1$ when the operation is a read and `ctx` and `addr` are both unchanged. The second constraint enforces that when either the context changed, the address changed, or the operation is a write, then `s_1` is set to $0$.
-
 
-To enforce the values of context ID, address, and clock cycle grow monotonically as described in the previous section, we define the following constraint.
+To enforce the values of context ID, word address, and clock cycle grow monotonically as described in the previous section, we define the following constraint.
 
 >$$
-\left(n_0 \cdot \Delta c + (1 - n_0) \cdot (n_1 \cdot \Delta a + (1 - n_1) \cdot \Delta i) \right) - (2^{16} \cdot d_1' + d_0') = 0 \text{ | degree} = 5
+f_{mem\_nl} \cdot \left(n_0 \cdot \Delta ctx + (1 - n_0) \cdot (n_1 \cdot \Delta a + (1 - n_1) \cdot \Delta clk) \right) - (2^{16} \cdot d_1' + d_0') = 0 \text{ | degree} = 8
 $$
```

Comment:
We should be able to use the definition of $f_{scw}$ to simplify this a bit (and maybe reduce degree):

$$
f_{meml} \cdot (n_0 \cdot \Delta ctx + (1 - n_0) \cdot n_1 \cdot \Delta a + f_{scw} \cdot \Delta clk) - (2^{16} \cdot d_1' + d_0')
$$

Reply:
Hmm, the above is marginally more simple - not sure it's much of an improvement in readability? How about

$$
f_{memnl} \cdot \left[ f_{scw} \cdot \Delta clk + (1 - f_{scw}) (n_0 \cdot \Delta ctx + (1 - n_0) \Delta a) \right] = 0
$$

Reply:
I'm fine with keeping it as is.